crates/op-rpc-types/src/op/transaction/mod.rs

```diff
@@ -0,0 +1,252 @@
+use alloy::{
+    consensus::{Signed, TxEip1559, TxEip2930, TxEip4844, TxEip4844Variant, TxEnvelope, TxLegacy},
+    rpc::types::eth::{AccessList, ConversionError, Signature},
+    serde as alloy_serde,
+};
+use alloy_primitives::{Address, Bytes, B256, U128, U256, U64};
+use serde::{Deserialize, Serialize};
+
+use self::{request::TransactionRequest, tx_type::TxType};
+
+pub mod receipt;
+pub mod request;
+pub mod tx_type;
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
+#[serde(rename_all = "camelCase")]
+pub struct Transaction {
```

Comment:
this type should also wrap the alloy transaction type and add the additional fields